src/Utilities/StdArrayHelpers.hpp

```diff
@@ -148,3 +150,21 @@ inline T magnitude(const std::array<T, 3>& a) noexcept {
   return sqrt(a[0] * a[0] + a[1] * a[1] + a[2] * a[2]);
 }
 //@}
+
+namespace std_array_helpers_detail {
+template <typename T, size_t Dim, typename F, size_t... Indices>
+auto map_array_impl(const std::array<T, Dim>& array, const F& f,
+                    const std::index_sequence<Indices...> /*meta*/) noexcept {
+  return std::array<std::decay_t<decltype(f(std::declval<T>()))>, Dim>{
+      {f(array[Indices])...}};
+}
+}  // namespace std_array_helpers_detail
+
+/// \ingroup UtilitiesGroup
+/// Applies a function to each element of an array, producing a new
+/// array of the results.
```

Comment:
It would be worth mentioning that `T` does not need to be default constructible in this case (unlike if one tried to use `std::transform`)